overlord/confdbstate/confdbstate.go

```diff
@@ -189,19 +189,29 @@ func GetTransactionToSet(ctx *hookstate.Context, st *state.State, view *confdb.V
 
 		return tx, nil, nil
 	}
-	// TODO: add early concurrency checks
+
+	txs, _, err := getOngoingTxs(st, account, schemaName)
+	if err != nil {
+		return nil, nil, fmt.Errorf("cannot access confdb view %s: cannot check ongoing transactions: %v", view.ID(), err)
+	}
+
+	if txs != nil && (txs.WriteTxID != "" || len(txs.ReadTxIDs) > 0) {
```

Comment:
do we want to add predicates like CatStartWriteTx and CanStartReadTx to transactions?

Reply:
I'm not sure, we don't add/set the read/write ongoing transactions in transaction.go (we do it towards the end of these exported functions), so it would be somewhat asymmetrical. I could add add the helpers and use them here though if that's fine?